File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    let minCnt = {};
    
    for(let i=0; i<keymap.length; i++){
        let key = keymap[i].split("");
        for(let j=0; j<key.length; j++){
            if(minCnt[key[j]] === undefined || minCnt[key[j]] > j+1){
                minCnt[key[j]] = j+1;
            }
        }
    }
        
        for(let i=0; i<targets.length; i++){
            let target = targets[i].split("");
            let sum =0;
            
            for(let char of target){
            if(minCnt[char] === undefined){
                sum = -1;
                break;
            }else{
                sum += minCnt[char];
            }
            }
                answer.push(sum);
        }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    for target in targets:
        target = list(target)
        minCnt = 0
        for ch in target:
            temp = []
            for key in keymap:
                if ch in key:
                    temp.append(key.index(ch)+1)
            if temp:
                minCnt += min(temp)
            else:
                minCnt = -1
                break
        answer.append(minCnt)
    return answer